src/main/java/org/apache/mesos/hdfs/state/LiveState.java

```diff
@@ -64,17 +64,26 @@ public void removeRunningTask(Protos.TaskID taskId) {
   public void updateTaskForStatus(Protos.TaskStatus status) {
     //Case of name node, update the task map
     if (status.getTaskId().getValue().contains(HDFSConstants.NAME_NODE_TASKID)) {
+      //If initializing the first NN or reconciling the first NN or bootstrapping the first NN
+      //set the status to initialized
       if (status.getMessage().equals(HDFSConstants.NAME_NODE_INIT_MESSAGE)
           || (currentAcquisitionPhase.equals(AcquisitionPhase.RECONCILING_TASKS)
-          && !isNameNode1Initialized())) {
+              && !isNameNode1Initialized())
```

Comment:
We currently call this method only when the task status is running. However, you are correct that in the case of reconciliation we are assuming the task is running and initialized. We want to use TaskStatus.STARTING (the node began running) and TaskStatus.RUNNING (the node initialized). Also, with health checks if the task does not initialize or starts throwing an error, it will be killed. 

I added a TODO explaining those cases to be added:

TODO (elingg) Use Starting Status when the task is running, but not initialized. Use running
    // status when the task is initialized so that we can differentiate during the reconciliation
    // phase. Also, add the health checks which will kill the task if it doesn't properly
    // initialize or if it reaches an error state.